mayastor/src/grpc/mayastor/v1.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{
+    core::Share,
+    grpc::{rpc_submit, GrpcClientContext, GrpcResult, Serializer},
+    lvs::{Error as LvsError, Lvs, PoolArgs},
+    subsys::PoolConfig,
+};
+use futures::FutureExt;
+use nix::errno::Errno;
+use std::{convert::TryFrom, fmt::Debug, time::Duration};
+use tonic::{Request, Response, Status};
+
+use rpc::mayastorv1::*;
+
+#[derive(Debug)]
+struct UnixStream(tokio::net::UnixStream);
+
+use ::function_name::named;
+use std::panic::AssertUnwindSafe;
+
+#[derive(Debug)]
+pub struct MayastorSvc {
+    name: String,
+    interval: Duration,
+    rw_lock: tokio::sync::RwLock<Option<GrpcClientContext>>,
+}
+
+#[async_trait::async_trait]
+impl<F, T> Serializer<F, T> for MayastorSvc
+where
+    T: Send + 'static,
+    F: core::future::Future<Output = Result<T, Status>> + Send + 'static,
```

Comment:
This would look more readable with a separate where clause for the `Output`:

```suggestion
    F: core::future::Future + Send + 'static,
    F::Output: Result<T, Status>,
```

Reply:
I guess Result is not a trait so the where clause is throwing `not a trait` error. Is there another way to use the where clause?

Reply:
ah, didn't think of that. :) nm.